File: 3주차/ATM.c
#include <stdio.h>

int main(void){
    int n;
    scanf("%d", &n);

    int array[1000];
    int i, j, key;
    for(i=0; i < n; i++){
        scanf("%d", &array[i]);
    }

    for(i=0; i<n-1; i++){
        j=i;
        while(j >=0 && array[j] > array[j+1]){
            key = array[j];
            array[j] = array[j+1];
            array[j+1] = key;
            j--;
        }
    }
    int sum=0, result=0;
    for(i=0;i<n;i++){
        sum+=array[i];
        result+=sum;
    }
    printf("%d", result);
}

File: 중간공부/ATM.py
n = int(input())

m=list(map(int, input().split()))
sum=0
result=0
m.sort()
for i in range(0, len(m)):#3 1 4 3 2
    sum=sum+m[i]
    result=result+sum

print(result)


# 5
# 3 1 4 3 2 -> 1 2 3 3 4
# 결과 : 32
# 시간을 단축시키기 위해서 오름차순 정렬을 해준다 (앞사람의 시간은 계속 누적되기 때문에)
# 1+3+6+9+13 = 32
